src/app/lib/utilities/getRemoteConfig/index.js

```diff
@@ -0,0 +1,45 @@
+import toggles from '#lib/config/toggles';
+import constructTogglesEndpoint from '#contexts/ToggleContext/utils/constructTogglesEndpoint';
+import nodeLogger from '#lib/logger.node';
+import { CONFIG_REQUEST_RECEIVED, CONFIG_FETCH_ERROR } from '#lib/logger.const';
+import getOriginContext from '#contexts/RequestContext/getOriginContext';
+
+const logger = nodeLogger(__filename);
+
+const getRemoteConfig = async (service, cache) => {
+  const environment = process.env.SIMORGH_APP_ENV || 'local';
+  const localToggles = toggles[environment];
+  if (!localToggles.enableFetchingToggles.enabled) {
+    return null;
+  }
+
+  const { origin } = getOriginContext();
+  const url = constructTogglesEndpoint(service, origin);
+
+  const cachedResponse = cache && cache.get(url);
+  if (cachedResponse) {
+    return cachedResponse;
+  }
+
+  try {
+    logger.info(CONFIG_REQUEST_RECEIVED, { url, service });
+    const response = await fetch(url, { headers: { origin } });
```

Comment:
I think you need to add a timeout here in case the endpoint goes down and requests back up to the config server, we did this in the include fetch: https://github.com/bbc/simorgh/blob/latest/src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js#L55 

This will prevent page requests timing out if the config service is unavailable.

Reply:
I think this is a good idea